### ISSUE TITLE
[Task] KubeApiClient watcher discovery improperly handles replica scaling

### ISSUE THREAD

**Issue by andrewazores** [cryostatio/cryostat#1038]
```bash
$ make sample_app # one replica, LOST and FOUND are emitted
$ oc scale --replicas=2 deployment/vertx-fib-demo # LOST and FOUND notifications are emitted for BOTH of the replicas - looks like a watch MODIFIED event was picked up
$ oc scale --replicas=1 deployment/vertx-fib-demo # LOST and FOUND notifications are emitted for ONE of the replicas - looks like a watch MODIFIED event was picked up
$ oc scale --replicas=0 deployment/vertx-fib-demo # no notification
$ oc scale --replicas=1 deployment/vertx-fib-demo # LOST and FOUND notifications are emitted again
$ oc scale --replicas=2 deployment/vertx-fib-demo # LOST and FOUND notifications are emitted for BOTH replicas
```

_Originally posted by @andrewazores in https://github.com/cryostatio/cryostat/pull/1037#issuecomment-1205883788_

**Comment by andrewazores** [comment]
After #1028 a naive, brute-force solution could be for the KubeApiClient to simply install watchers on various resources (Endpoints, apps/Deployments, ReplicationControllers, Pods, discovery/v1/EndpointSlice, Services, Routes, etc). When any of these changes, perform a full re-query of the discovery tree and update the DiscoveryStorage. The DiscoveryStorage implementation already has logic to detect the changed targets and emit corresponding LOST/FOUND events.

**Comment by andrewazores** [comment]
Worth mentioning that our current approach of using an Endpoints watcher seems like it should have been the right approach, though:

https://kubernetes.io/docs/concepts/services-networking/service/#cloud-native-service-discovery

**Comment by andrewazores** [comment]
https://engineering.dollarshaveclub.com/kubernetes-fixing-delayed-service-endpoint-updates-fd4d0a31852c perhaps relevant

**Comment by andrewazores** [comment]
https://codeburst.io/kubernetes-watches-by-example-bc1edfb2f83

**Comment by andrewazores** [comment]
Fixed in #1037
